### PR TITLE
[SID-1184] Granular auth rules: authentication on page and path level

### PR DESCRIPTION
**Pull request by ikovic**
Added two new ways of configuring authentication:
- per page basis using front matter
- per path basis using `docusaurus.config.js`
- both ways allow groups to be specified

### Todos:
- [x] Update the README
- [x] Write docs
- [x] Implement an example in the demo app

### CODE REVIEW

File: packages/docusaurus-theme-slashid/src/domain.test.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        // @ts-expect-error cannot import User for some reason
+        shouldPathRender("/foo", [{ path: "/foo", groups: ["bar"] }], {
+          getGroups: () => [],
+        })
+      ).toBe(false);
+    });
+
+    test("should return false if a user is not present and the provided regex matches", () => {
+      expect(
+        shouldPathRender("/foo/bar", [{ path: /^\/foo/ }], undefined)
```
Comment (jakewhelan) [inline packages/docusaurus-theme-slashid/src/domain.test.ts:57]: Should we support globs?

We could use something like [micromatch](https://www.npmjs.com/package/micromatch) to do it cheaply

**Review by jakewhelan — APPROVED** [review]
Looks good to me!

~Is there any example of how to use the feature? I don't see any update to README.md, but also I don't think there is an example in `apps/demo` unless I'm missing something~ 🤔

Edit:

Nevermind 😂 🤦 🤦 🤦 

image

**Comment by jakewhelan** [comment]
Nevermind 😂 🤦 🤦 🤦 

image

File: packages/docusaurus-theme-slashid/src/domain.test.ts (hunk 1 above)
Comment (ikovic) [reply packages/docusaurus-theme-slashid/src/domain.test.ts:57]: Ahh I started from globs but then when researching libraries I couldn't find anything really small - even micromatch adds 35kb minified. Globs would surely look better though!

Do you think it's worth the cost? I tested it with a couple of simple regex and it worked OK, but globs would be more intuitive.

Comment (jakewhelan) [reply packages/docusaurus-theme-slashid/src/domain.test.ts:57]: I'm biased but I would say it's worth it, regex is scary

Comment (paulo-raca) [reply packages/docusaurus-theme-slashid/src/domain.test.ts:57]: Maybe a glob-to-regex library is smaller in size than a standalone matcher? E.g., https://www.npmjs.com/package/glob-to-regexp

Comment (ikovic) [reply packages/docusaurus-theme-slashid/src/domain.test.ts:57]: @paulo-raca this is interesting, it's under 1kB. Let me test it out!

Comment (ikovic) [reply packages/docusaurus-theme-slashid/src/domain.test.ts:57]: Done!

**Review by kaspermroz — APPROVED** [review]

**Pull request merged**